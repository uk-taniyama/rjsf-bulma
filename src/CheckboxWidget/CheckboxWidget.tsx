import { WidgetProps } from "@rjsf/utils";
import { FieldGroup, FieldControl } from "../ui";

const CheckboxWidget = (props: WidgetProps) => {
  const {
    id,
    value,
    required,
    disabled,
    readonly,
    label,
    schema,
    autofocus,
    onChange,
    onBlur,
    onFocus,
  } = props;

  const _onChange = ({
    target: { checked },
  }: React.FocusEvent<HTMLInputElement>) => onChange(checked);
  const _onBlur = ({
    target: { checked },
  }: React.FocusEvent<HTMLInputElement>) => onBlur(id, checked);
  const _onFocus = ({
    target: { checked },
  }: React.FocusEvent<HTMLInputElement>) => onFocus(id, checked);

  const desc = label || schema.description;
  return (
    <FieldGroup>
      <FieldControl>
        <label>
          <input
            id={id}
            name={id}
            checked={typeof value === "undefined" ? false : value}
            required={required}
            disabled={disabled || readonly}
            autoFocus={autofocus}
            onChange={_onChange}
            type="checkbox"
            onBlur={_onBlur}
            onFocus={_onFocus}
          />
          {desc}
        </label>
      </FieldControl>
    </FieldGroup>
  );
};

export default CheckboxWidget;
