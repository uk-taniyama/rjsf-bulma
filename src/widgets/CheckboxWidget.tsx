// FIXME checkboxを横並びにしている.....
// FIXME スタイルを強引に当てている.....
import type { FC } from "react";
import type { WidgetProps } from "@rjsf/utils";
import { FieldControl } from "../ui";

const CheckboxWidget: FC<WidgetProps> = (props) => {
  const {
    id,
    value,
    required,
    disabled,
    readonly,
    label,
    schema,
    autofocus,
    onChange,
    onBlur,
    onFocus,
  } = props;

  const _onChange = ({
    target: { checked },
  }: React.FocusEvent<HTMLInputElement>) => onChange(checked);
  const _onBlur = ({
    target: { checked },
  }: React.FocusEvent<HTMLInputElement>) => onBlur(id, checked);
  const _onFocus = ({
    target: { checked },
  }: React.FocusEvent<HTMLInputElement>) => onFocus(id, checked);

  const desc = label || schema.description;
  return (
    <>
      <FieldControl>
        <label className="checkbox is-small is-size-7 mr-1">
          <input
            id={id}
            name={id}
            type="checkbox"
            className="mr-1"
            style={{ verticalAlign: "middle" }}
            checked={value == null ? false : value}
            required={required}
            disabled={disabled || readonly}
            autoFocus={autofocus}
            onChange={_onChange}
            onBlur={_onBlur}
            onFocus={_onFocus}
          />
          {desc}
        </label>
      </FieldControl>
    </>
  );
};

export default CheckboxWidget;
