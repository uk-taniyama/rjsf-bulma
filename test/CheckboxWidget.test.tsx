import React from "react";
import CheckboxWidget from "../src/widgets/CheckboxWidget";
import renderer from "react-test-renderer";
import { makeWidgetMockProps } from "./helpers/createMocks";

describe("CheckboxWidget", () => {
  test("simple", () => {
    const tree = renderer
      .create(<CheckboxWidget {...makeWidgetMockProps({})} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
