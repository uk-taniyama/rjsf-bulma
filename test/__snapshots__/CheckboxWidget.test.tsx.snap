// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxWidget simple 1`] = `
<div
  className="control"
>
  <label
    className="checkbox"
  >
    <input
      autoFocus={true}
      checked="value"
      disabled={true}
      id="_id"
      name="_id"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={true}
      type="checkbox"
    />
    Some simple label
  </label>
</div>
`;
